src/main/java/ch/wisv/areafiftylan/security/authentication/CurrentUserServiceImpl.java

```diff
@@ -128,49 +146,45 @@ public boolean canAccessOrder(Object principal, Long orderId) {
             User user = (User) principal;
             // Return true if the order is owned by the user, or the user is an admin
             return order.getUser().getEmail().equals(user.getEmail()) ||
-                    user.getAuthorities().contains(Role.ROLE_ADMIN);
-        } else {
-            return false;
+                    isAdmin(user);
         }
+        return false;
     }
 
     @Override
     public boolean isTicketOwner(Object principal, Long ticketId) {
         if (principal instanceof UserDetails) {
             User user = (User) principal;
             return ticketService.getTicketById(ticketId).getOwner().equals(user);
-        } else {
-            return false;
         }
+        return false;
     }
 
     @Override
     public boolean canReserveSeat(Object principal, Long ticketId) {
         if (principal instanceof UserDetails) {
             User user = (User) principal;
 
-            Ticket ticket = ticketRepository.findOne(ticketId);
-            if (ticket.getOwner().equals(user) || user.getAuthorities().contains(Role.ROLE_ADMIN)) {
+            User owner = ticketRepository.findOne(ticketId).getOwner();
+            if (owner.equals(user) || isAdmin(user)) {
                 return true;
             }
 
             Collection<Team> userTeams = teamService.getTeamByCaptainId(user.getId());
 
             // For each set of members in a team, check if the owner of the ticket is one of them.
-            return userTeams.stream().map(Team::getMembers).anyMatch(members -> members.contains(ticket.getOwner()));
-        } else {
-            return false;
+            return userTeams.stream().map(Team::getMembers).anyMatch(members -> members.contains(owner));
         }
+        return false;
     }
 
     @Override
     public boolean hasAnyTicket(Object principal) {
```

Comment:
Since we're not restricting the creation of teams to ticketholders, this is no longer used